Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Utils/IndexedDataVector.h

```diff
@@ -27,31 +27,49 @@ namespace AZ::Render
         static constexpr IndexType NoFreeSlot = std::numeric_limits<IndexType>::max();
         IndexType m_firstFreeSlot = NoFreeSlot;
 
+        //! Clears all data and resets to initial state.
         void Clear();
+
+        //! Creates a new entry, default-constructs it, and returns an index that references it.
         IndexType GetFreeSlotIndex();
+
+        //! Destroys the data referenced by index and frees that index for future use.
         void RemoveIndex(IndexType index);
+
+        //! Destroys the data and related index by using a pointer to the data itself.
         void RemoveData(DataType* data);
 
+        //! Returns a reference to the data using the provided index.
         DataType& GetData(IndexType index);
         const DataType& GetData(IndexType index) const;
+
+        //! Returns a count of how many items are stored in the IndexedDataVector
         size_t GetDataCount() const;
 
+        //! Returns a reference to the internal data vector.
+        //! This vector should not be altered by calling code or the IndexedDataVector will be corrupted
         AZStd::vector<DataType>& GetDataVector();
         const AZStd::vector<DataType>& GetDataVector() const;
+        
+        //! Returns a reference to the internal vector.
+        const AZStd::vector<IndexType>& GetDataToIndexVector() const;
 
-        AZStd::vector<IndexType>& GetIndexVector();
-        const AZStd::vector<IndexType>& GetIndexVector() const;
-
+        //! Returns the offset into the internal data vector for a given index.
         IndexType GetRawIndex(IndexType index) const;
+
+        //! Returns the public index for data given its pointer.
```

Comment:
The part I found confusing at first was that m_indices values might be physical indexes when the data is present, or might be logical indexes into itself to form the linked list of free spots.
I think the comment for that member could still be expanded a bit